api/python/provisioner/commands/set_swupgrade_repo.py

```diff
@@ -333,6 +376,54 @@ def dynamic_validation(self, params: inputs.SWUpgradeRepo, targets: str):  # noq
 
         return repo.metadata
 
+    @staticmethod
+    def _setup_python_index(repo: inputs.SWUpgradeRepo):
+        """
+        Setup Python index
+
+        Parameters
+        ----------
+        repo: inputs.SWUpgradeRepo
+            SW upgrade repository parameters
+
+        Returns
+        -------
+
+        """
+        def setup_default():
+            _config = ConfigParser()
+
+            _config['global'] = {
+                'timeout': '60',
+                'index-url': f'file:///{python_index_path}'
+            }
+            return _config
+
+        iso_mount_dir = repo.target_build / repo.release
+        python_index_path = iso_mount_dir / CORTX_PYTHON_ISO_DIR
+        if python_index_path in iso_mount_dir.iterdir():
+            if not Path(PIP_CONFIG_FILE).exists():
+                config = setup_default()
+
+            else:
+                config = ConfigParser()
+                config.read(PIP_CONFIG_FILE)
+                extra_index_url = config['global'].get('extra-index-url', None)
+                if extra_index_url:
+                    if str(python_index_path) not in extra_index_url:
+                        # NOTE: if is needed to avoid index duplication
+                        config['global']['extra-index-url'] = (
+                            f"{extra_index_url}\n"
+                            f"file://{python_index_path.absolute()}"
```

Comment:
please check the comment below regarding `absolute`

Reply:
Done